### PR TITLE
DOC: instructions on installing matplotlib for dev

### PR DESCRIPTION
**Pull request by eyurtsev**
Added instructions on installing matplotlib for dev for issue (https://github.com/matplotlib/matplotlib/issues/3959)

NOTE: I renamed one of the URLs. git_development -> setting_up_for_development 

let me know if this is a problem.

### CODE REVIEW

File: doc/devel/gitwash/matplotlib_for_dev.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+After obtaining a local copy of the matpotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell:
+
+::
+    
+    python setup.py develop
+
+This installs matplotlib for development (i.e., builds everything and places the
+symbolic links back to the source code).
```
Comment (fariza) [inline doc/devel/gitwash/matplotlib_for_dev.rst]: As noted in ` faq/installing_faq.rst` this command has to be called everytime that the c files change

**Comment by JanSchulz** [comment]
There is now also some information how to setup on windows in the main directory.

**Comment by NelleV** [comment]
This PR is a bit old, but it would be a great addition to the documentation. @eyurtsev are you still interested in moving forward with this PR?

**Comment by eyurtsev** [comment]
I've been out of the loop with matplotlib for a while, so if anyone would like to commandeer that'd be great -- alternatively, I can abandon -- whichever is most convenient. :)

File: doc/devel/gitwash/matplotlib_for_dev.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+===================================================
+    Install for matplotlib source for development
+===================================================
+
+After obtaining a local copy of the matpotlib source code (:ref:`set-up-fork`),
+navigate to the matplotlib directory and run the following in the shell:
+
+::
+    
+    python setup.py develop
```
Comment (tacaswell) [inline doc/devel/gitwash/matplotlib_for_dev.rst]: This should probably be changed to `pip install -v -e .`

**Comment by Kojoley** [comment]
We already have this here [doc/faq/installing_faq.rst#source-install-from-git](https://github.com/matplotlib/matplotlib/blob/32eb2c5130548cf78d538f4eb578cb59bbd4c4b7/doc/faq/installing_faq.rst#source-install-from-git)

**Comment by NelleV** [comment]
Hi @eyurtsev 
I can take over. I'll probably have to recreate a PR.
I'll keep you posted.
Cheers,
N

**Comment by NelleV** [comment]
I created a new PR with these changes + the fixes. I'm thus closing this one.
Thanks @eyurtsev for the original patch and the feedback!

**Pull request closed without merging**